### PR TITLE
feat: Add vault implementation

### LINKED ISSUE #1183: Implement Add Vault
https://app.abstract.com/projects/9ec92568-60a2-4fee-8215-5aa997d52428/branches/dc10bcee-d9a1-44f9-b644-8f843c6f1a19/commits/762f23dd8207318f36641e723d6d7dcf1d9c6451/files/95d5757f-f10e-43f6-ad50-0faa236217fd/layers/86D0CB0A-95CC-4F2A-962A-88DEC9807A3E?collectionId=cc5337ea-21d1-4d7d-8d0f-97b0e5414ebb&collectionLayerId=c8c87afd-1cba-4e18-93b0-4d54bbbe1b50

### LINKED ISSUE #1190: Refactor Vault cards

### PR DESCRIPTION
**Pull request by talwinder50**
https://user-images.githubusercontent.com/7862595/141301487-04bfa7b3-ee22-49f0-a13a-b5516064f400.mov

https://user-images.githubusercontent.com/7862595/141301577-2b064e1b-a8ec-49d5-826c-6452beabb92a.mov

Screen Shot 2021-11-11 at 6 43 03 AM
Screen Shot 2021-11-11 at 6 42 53 AM
Screen Shot 2021-11-11 at 6 42 27 AM
Screen Shot 2021-11-11 at 6 42 37 AM

closes #1183  #1209 #1210

Signed-off-by: talwinder50 <talwinderkaur50@gmail.com>

### CODE REVIEW

File: cmd/wallet-web/src/components/Modal/AddVault.vue [hunk 1]
```diff
@@ -0,0 +1,140 @@
+<template>
```
Comment (rolsonquadras) [inline cmd/wallet-web/src/components/Modal/AddVault.vue:1]: missing file licence header.

**Review by rolsonquadras — APPROVED** [review]

File: cmd/wallet-web/src/components/InputField/InputField.vue [hunk 2]
```diff
@@ -1,23 +1,22 @@
-<!--
- * Copyright SecureKey Technologies Inc. All Rights Reserved.
- *
- * SPDX-License-Identifier: Apache-2.0
--->
+/* Copyright SecureKey Technologies Inc. All Rights Reserved. SPDX-License-Identifier: Apache-2.0 */
```
Comment (birtony) [inline cmd/wallet-web/src/components/InputField/InputField.vue:1]: This is not an expected change, is it?

File: cmd/wallet-web/src/components/InputField/InputField.vue [hunk 3]
```diff
@@ -1,23 +1,22 @@
...
     />
     <label :for="'input-' + label" class="input-label">{{ label }}</label>
     <span class="input-helper">{{ helperMessage }}</span>
-    <div class="fader" />
-    <!-- TODO: use inline svg instead once https://github.com/trustbloc/edge-agent/issues/816 is fixed -->
-    <img src="@/assets/img/danger-icon.svg" />
+    <div class="fader">
+      <!-- TODO: use inline svg instead once https://github.com/trustbloc/edge-agent/issues/816 is fixed -->
+      <img src="@/assets/img/danger-icon.svg" />
+    </div>
```
Comment (birtony) [inline cmd/wallet-web/src/components/InputField/InputField.vue:18]: This change does not seem to be correct. The danger icon is not supposed to be placed on the fader, it's supposed to be to the right from it.

File: cmd/wallet-web/src/components/Modal/AddVault.vue [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    receivedVaultName(data) {
+      this.vaultName = data;
+    },
+    addVault() {
+      // Integration with collections API.
+      let { user, token } = this.getCurrentUser().profile;
+      this.username = this.getCurrentUser().username;
+      let collectionManager = new CollectionManager({ agent: this.getAgentInstance(), user });
+      const name = this.vaultName;
+      let id = collectionManager.create(token, { name });
```
Comment (birtony) [inline cmd/wallet-web/src/components/Modal/AddVault.vue:134]: Can we make these `const`?

File: cmd/wallet-web/src/pages/Vaults.vue [hunk 5]
```diff
@@ -63,6 +70,27 @@ export default {
...
       this.credentialsFound = Object.keys(contents).length;
     },
+    fetchAllVaults: async function (token, collectionManager) {
+      const { contents } = await collectionManager.getAll(token);
+      console.log(`found ${Object.keys(contents).length} vaults`);
+
+      const collections = Object.keys(contents);
+
+      collections.map(async (id) => {
+        let vault = await collectionManager.get(token, id);
```
Comment (birtony) [inline cmd/wallet-web/src/pages/Vaults.vue:80]: And this one too?

File: cmd/wallet-web/src/components/InputField/InputField.vue (hunk 3 above)
Comment (talwinder50) [reply cmd/wallet-web/src/components/InputField/InputField.vue]: fixed as original

File: cmd/wallet-web/src/components/InputField/InputField.vue (hunk 2 above)
Comment (talwinder50) [reply cmd/wallet-web/src/components/InputField/InputField.vue]: dont know how this got edited probably by prettier i fixed though

**Review by birtony — APPROVED** [review]

File: cmd/wallet-web/src/pages/Vaults.vue [hunk 6]
```diff
@@ -63,6 +70,27 @@ export default {
       const { contents } = await getCredential;
       this.credentialsFound = Object.keys(contents).length;
     },
+    fetchAllVaults: async function (token, collectionManager) {
+      const { contents } = await collectionManager.getAll(token);
+      console.log(`found ${Object.keys(contents).length} vaults`);
+
+      const collections = Object.keys(contents);
+
+      collections.map(async (id) => {
```
Comment (sudeshrshetty) [inline cmd/wallet-web/src/pages/Vaults.vue:79]: You have to use "array.Map()" when you are building new array by using underlying array as reference.

In this case, you should use array.ForEach() or `this.vaults = collections.map(.....)` where collection map function returns vault (`return vault`)

Comment (sudeshrshetty) [reply cmd/wallet-web/src/pages/Vaults.vue:79]: & `getAll` already returned you all the vaults in key-value pairs, why you have to hit the store again?

**Pull request merged**